app/src/main/java/se/nullable/flickboard/KeyboardService.kt

```diff
@@ -402,11 +404,26 @@ class KeyboardService : InputMethodService(), LifecycleOwner, SavedStateRegistry
 
                                 Action.ToggleEmojiMode ->
                                     emojiMode = !emojiMode
+
+                                Action.EnableVoiceMode -> {
+                                    voiceMode = true
```

Comment:
This isn't really a _state_ of _this_ keyboard, so you can just do the switcharoo in here instead.

Reply:
I did try that at first, but ran into issues with this not being a `@Composable` method, which I needed to fetch the InputMethodManager from the current Context. Is it kosher to just add the annotation to onCreateInputView, or is there some other means of invoking the Composable method from here?

Reply:
The `KeyboardService` is already a `Context`.

Reply:
That was a big missing piece and simplified a lot of this implementation, thanks